src/ranch_acceptors_sup.erl

```diff
@@ -106,9 +106,14 @@ hide_socket_opts([{password, _}|SocketOpts]) ->
 hide_socket_opts([SocketOpt|SocketOpts]) ->
 	[SocketOpt|hide_socket_opts(SocketOpts)].
 
-format_error(no_cert) ->
+format_error(_, no_cert) ->
 	"no certificate provided; see cert, certfile, sni_fun or sni_hosts options";
-format_error(reuseport_local) ->
+format_error(_, reuseport_local) ->
 	"num_listen_sockets must be set to 1 for local sockets";
-format_error(Reason) ->
-	inet:format_error(Reason).
+format_error(Transport, Reason) ->
+	case erlang:function_exported(Transport, format_error, 1) of
```

Comment:
Add a comment saying this will be required in 3.0, and for now let's consider/document it as optional.